### PR TITLE
Refactor hash task

### PR DESCRIPTION
**Pull request by Trial97**
Refactor the mess of hashUtilities and make the task abortable and async
And made provider capabilities a namespace instead of a class(which makes more sense to me).

### CODE REVIEW

**Review by Ryex — APPROVED** [review]
This looks so much cleaner now

File: launcher/modplatform/helpers/HashUtils.h [hunk 1]
```diff
@@ -1,67 +1,51 @@
...
-    void executeTask() override;
-
-    QStringList getHashTypes();
-    bool useHashType(QString type);
-
-   private:
-    ModPlatform::ResourceProvider provider;
-    QString hash_type;
+    QFuture<QString> m_zip_future;
+    QFutureWatcher<QString> m_zip_watcher;
```
Comment (TheKodeToad) [inline launcher/modplatform/helpers/HashUtils.h]: zip?

Comment (Trial97) [reply launcher/modplatform/helpers/HashUtils.h]: dam forgot to rename

**Review by TheKodeToad — APPROVED** [review]

**Comment by Trial97** [comment]
This was a low prio but moving it to 9.0 due to approvals

**Pull request merged**